java/clients/src/main/java/sleeper/clients/QueryWebSocketClient.java

```diff
@@ -71,41 +98,56 @@ protected void init(TableProperties tableProperties) {
 
     @Override
     protected void submitQuery(TableProperties tableProperties, Query query) {
-        Client client = null;
         try {
             Instant startTime = Instant.now();
-            client = new Client(URI.create(apiUrl), query, querySerDe);
-            while (!client.isQueryComplete()) {
+            client.startQuery(query);
+            while (!client.hasQueryFinished()) {
                 Thread.sleep(500);
```

Comment:
This could be stubbed to stop the unit tests from waiting. It looks like the waiting is never tested though, since in all the tests the query is finished after startQuery is called. Can we add tests for this? This could be a separate issue if you prefer.

Reply:
Thanks, i've raised the following issue for this: https://github.com/gchq/sleeper/issues/2161